app/controllers/preferences_controller.rb

```diff
@@ -5,4 +5,24 @@ def index
     @preferences = current_user.preferences
     @pagy, @records = pagy(@preferences)
   end
+
+  def new
+    @preference = current_user.preferences.build
+  end
+
+  def create
+    @preference = current_user.preferences.build(preference_params)
+
+    if @preference.save
+      redirect_to preferences_path
```

Comment:
What about adding a success message when you return to index?

Reply:
Added